core/trino-main/src/main/java/io/trino/sql/planner/PlanFragment.java

```diff
@@ -204,6 +204,26 @@ public PlanFragment withoutEmbeddedJsonRepresentation()
                 this.activeCatalogs);
     }
 
+    public PlanFragment withPartitioning(PartitioningHandle partitioning)
+    {
+        if (jsonRepresentation.isEmpty()) {
```

Comment:
This check probably should not be here. I think it's only needed for `withoutEmbeddedJsonRepresentation()`

Reply:
@arhimondr should this check be removed?

Reply:
Yeah, this check seems to be harmless but redundant.

Reply:
Removed.